Tribler/Utilities/Instance2Instance.py

```diff
@@ -29,11 +30,16 @@ class I2IFactory(Factory):
     def __init__(self, callback):
         self.callback = callback
 
+
 class Instance2InstanceServer():
 
-    def __init__(self, i2iport, callback):
-        endpoint = TCP4ServerEndpoint(reactor, i2iport, interface="127.0.0.1")
-        endpoint.listen(I2IFactory(callback))
+    def __init__(self, i2iport):
+        self.endpoint = TCP4ServerEndpoint(reactor, i2iport, interface="127.0.0.1")
+
+    @inlineCallbacks
+    def start(self, callback):
+        self.listenport = yield self.endpoint.listen(I2IFactory(callback))
+        returnValue(self.listenport)
```

Comment:
There's no stop method and you need to update Tribler to start and stop the instance2instance stuff.